### PR TITLE
[INLONG-9893][Common] Optimized code InLongMsg

### PR DESCRIPTION
**Pull request by beatCao**
Title Name: [INLONG-9893] Optimized code InLongMsg.java

https://github.com/apache/inlong/issues/9893

Motivation
Explain here the context, and why you're making that change. What is the problem you're trying to solve.

Modifications
Describe the modifications you've done.

### CODE REVIEW

File: inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsg.java [hunk 1]
```diff
@@ -246,13 +246,12 @@ public boolean addMsg(String attr, ByteBuffer data) {
...
-        DataBuffer outputBuffer = attr2MsgBuffer.get(attr);
-        if (outputBuffer == null) {
-            outputBuffer = new DataBuffer();
-            attr2MsgBuffer.put(attr, outputBuffer);
+        DataBuffer outputBuffer = attr2MsgBuffer.computeIfAbsent(attr, key -> {
+            DataBuffer newBuffer = new DataBuffer();
             // attrlen + utflen + meglen + compress
-            this.datalen += attr.length() + 2 + 4 + 1;
-        }
+            this.datalen += key.length() + 2 + 4 + 1;
```
Comment (aloyszhang) [inline inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsg.java:252]: This line is not the logic of the `mappingFunction`

**Review by aloyszhang — CHANGES_REQUESTED** [review]
Thanks for your contribution. 
Just left a comment, PTAL.

File: inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsg.java (hunk 1 above)
Comment (beatCao) [reply inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsg.java]: > This line is not the logic of the `mappingFunction`
Problem fixed

**Comment by dockerzhang** [comment]
Run 'mvn spotless:apply' to fix the violations.

**Comment by beatCao** [comment]
> Run 'mvn spotless:apply' to fix the violations.



> Run 'mvn spotless:apply' to fix the violations.

**Comment by beatCao** [comment]
> Run 'mvn spotless:apply' to fix the violations.

Problem fixed

File: inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsg.java [hunk 2]
```diff
@@ -246,13 +246,12 @@ public boolean addMsg(String attr, ByteBuffer data) {
...
-        if (outputBuffer == null) {
-            outputBuffer = new DataBuffer();
-            attr2MsgBuffer.put(attr, outputBuffer);
+        DataBuffer outputBuffer = attr2MsgBuffer.computeIfAbsent(attr, key -> {
+            DataBuffer newBuffer = new DataBuffer();
             // attrlen + utflen + meglen + compress
             this.datalen += attr.length() + 2 + 4 + 1;
-        }
+            return newBuffer;
+        });
```
Comment (aloyszhang) [inline inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsg.java:254]:
```java
DataBuffer outputBuffer = attr2MsgBuffer.computeIfAbsent(attr, key -> {
            DataBuffer newBuffer = new DataBuffer();
            // attrlen + utflen + meglen + compress
            this.datalen += attr.length() + 2 + 4 + 1;
            return newBuffer;
        });
```
computeIfAbsent should only create a new DataBuffer if the mapping for attr does not exist.
These two lines should move out of the mappingFunction.
```java
            // attrlen + utflen + meglen + compress
            this.datalen += attr.length() + 2 + 4 + 1;
```
Suggestion:
```java
        attr2MsgBuffer.computeIfAbsent(attr, k-> new DataBuffer());
         // attrlen + utflen + meglen + compress
        this.datalen += attr.length() + 2 + 4 + 1;
```

**Comment by dockerzhang** [comment]
@beatCao you could keep the PR open, you do not need to open and close it again.

**Review by aloyszhang — APPROVED** [review]
LGTM

**Comment by aloyszhang** [comment]
image
No more review is required from one committer or PMC member if he has approved before.

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**